src/screens/ClipsScreen.tsx

```diff
@@ -298,9 +304,45 @@ export class ClipsScreen extends React.Component<Props, State> {
   }
 
   _handleSearchBarTextQuery = () => {
-    const queryFrom = PV.Filters._allPodcastsKey
-    const keepSearchTitle = true
-    this.handleSelectFilterItem(queryFrom, keepSearchTitle)
+    const { searchBarText, tempQueryEnabled } = this.state
+
+    if (!searchBarText) {
+      this._handleRestoreSavedQuery()
+    } else {
+      const { queryFrom, querySort } = this.state
+      const tempQueryObj = !tempQueryEnabled
+        ? {
+            tempQueryEnabled: true,
+            tempQueryFrom: queryFrom,
+            tempQuerySort: querySort
+          }
+        : {
+            tempQueryEnabled: true,
+            tempQueryFrom: this.state.tempQueryFrom,
+            tempQuerySort: this.state.tempQuerySort
```

Comment:
These are already pull out from state, do you need to access them through state here as well?

Reply:
The only reason they're here tbh is to make TypeScript stop telling me I'm passing in an invalid object format...

I can clean it up a little by using an "any" 😬